IANotificationsTestProject/In App Notifications/NoticePresenter.swift

```diff
@@ -0,0 +1,95 @@
+import UIKit
+
+class NoticePresenter {
+    
+    // MARK: Properties
+    var noticeView: NoticeView?
+    var noticeTopConstraint: NSLayoutConstraint?
+
+    var isPresenting: Bool {
+        noticeView != nil
+    }
+
+    // MARK: Presenting/Dismissing Methods
+
+    func presentNoticeView(_ noticeView: NoticeView, completion: @escaping () -> Void) {
+        guard let keyWindow = getKeyWindow() else {
+            return
+        }
+        self.noticeView = noticeView
+        keyWindow.addSubview(noticeView)
+
+        noticeTopConstraint = noticeView.topAnchor.constraint(equalTo: keyWindow.bottomAnchor)
+        noticeTopConstraint?.isActive = true
+        NSLayoutConstraint.activate([
+            noticeView.centerXAnchor.constraint(equalTo: keyWindow.centerXAnchor),
+            noticeView.widthAnchor.constraint(equalToConstant: CGFloat(275)),
+            noticeView.heightAnchor.constraint(equalToConstant: 50)
+        ])
+
+        keyWindow.layoutIfNeeded()
+        displayNotificationView {
+            completion()
+        }
+    }
+
+    private func displayNotificationView(completion: @escaping () -> Void) {
+        guard let keyWindow = getKeyWindow(),
+              let noticeView = noticeView else {
+            return
+        }
+
+        noticeTopConstraint?.isActive = false
+        let newConstraint = noticeView.bottomAnchor.constraint(equalTo: keyWindow.bottomAnchor, constant: Constants.notificationBottomMargin)
+        newConstraint.isActive = true
+
+        let delay = noticeView.action == nil ? Times.waitShort : Times.waitLong
+
+        UIView.animate(withDuration: Times.animationTime) {
+            keyWindow.layoutIfNeeded()
```

Comment:
This one is probably risky: the entire view hierarchy will be laid out (I've seen, in the past, unexpected animations caused by similar code!).

One potential approach is to have a UIView on top of everything, and have the Notice attached to it (so that you can just do `containerView.layoutIfNeeded`